### PR TITLE
Neon uint16

### PR DESCRIPTION
**Pull request by martinRenou**

### CODE REVIEW

File: include/xsimd/types/xsimd_neon_uint16.hpp [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+
+    inline uint16_t batch<uint16_t, 8>::operator[](std::size_t index) const
+    {
+        return m_value[index];
+    }
+
+    namespace detail
+    {
+
+#define XSIMD_INLINE_UNROLL_OP(op, i)  \
```
Comment (wolfv) [inline include/xsimd/types/xsimd_neon_uint16.hpp]: are weable to use the template unroller here?

File: include/xsimd/types/xsimd_neon_uint16.hpp [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+            {
+                return uint16x8_t{
+                    XSIMD_INLINE_UNROLL_OP_8(/)
+                };
+            }
+
+            static batch_type mod(const batch_type& lhs, const batch_type& rhs)
+            {
+                return uint16x8_t{
+                    XSIMD_INLINE_UNROLL_OP_8(%)
```
Comment (wolfv) [inline include/xsimd/types/xsimd_neon_uint16.hpp]: maybe not! :)

Comment (martinRenou) [reply include/xsimd/types/xsimd_neon_uint16.hpp]: It is a bit confusing, but `XSIMD_INLINE_UNROLL_OP_8` is for 8 elements, not for int8. And we have 8 elements of uint16 in the batch here.

Comment (martinRenou) [reply include/xsimd/types/xsimd_neon_uint16.hpp]: Or maybe I don't understand the problem you're seeing?

Comment (martinRenou) [reply include/xsimd/types/xsimd_neon_uint16.hpp]: Maybe we should rename the macro `XSIMD_INLINE_UNROLL_OP_UINT16x8` so that it's not confusing anymore

Comment (wolfv) [reply include/xsimd/types/xsimd_neon_uint16.hpp]: Sorry, I wasn't clear. I was referring to my statement above regarding usage of the meta-unroller. 
But I think we can't use it in a initializer, so a MACRO is necessary.

Comment (martinRenou) [reply include/xsimd/types/xsimd_neon_uint16.hpp]: I made a template unroller like you suggested :) better than the macro

Comment (wolfv) [reply include/xsimd/types/xsimd_neon_uint16.hpp]: looks great!

**Pull request merged**